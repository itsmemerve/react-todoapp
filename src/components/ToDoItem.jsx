import React, { useState } from 'react'
import {  Tag, List, Button, Popconfirm,  Select, Modal, Form, Row, Col, Input ,message } from 'antd';
import { EditOutlined } from '@ant-design/icons';
const { Option } = Select;

const Todo = ({ todo, onTodoRemoval,
  onTodoToggle}) => {

  const [modalShow, setModalShow] = useState(false);
  const [form] = Form.useForm();

  const statusList = [ {
    id: 0,
    name: "Not Started"
  },
  {
    id:1,
    name: "In Progress",
  }, {
    id: 2,
    name: "Completed"
  }];

  const priorityList = [ {
    id: 0,
    name: "Low"
  },
  {
    id: 1,
    name: "Medium"
    }, {
    id: 2,
    name: "High"
  }];

  const onbuttonclick = (e) => {
    setModalShow(true);
  }
  
  const handleCancel = () => {
    setModalShow(false);
  };

  const handleOk = () => {
    setModalShow(false);
  };


  const handleOnFinish = () => {
    if(form.getFieldValue('name') === ""){
      message.warn("Name is required!");
      return;
    }
    onTodoToggle({
      Id: todo.id,
      Name: form.getFieldValue('name') === undefined ? todo.name : form.getFieldValue('name') ,
      Status: form.getFieldValue('status') === undefined ? todo.status : form.getFieldValue('status'),
      Priority: form.getFieldValue('priority') === undefined ? todo.priority :  form.getFieldValue('priority')
    });
  };

  return (
    <List.Item
      actions={[
        <Popconfirm key={`${todo.id}-delete-popup`} title="Are you sure you want to delete?" onConfirm={() => { onTodoRemoval(todo);}}>
          <Button className="remove-todo-button" type="primary" danger  key={`${todo.id}-deletbutton`}>X</Button>
        </Popconfirm>,
        <Button onClick={(e) => onbuttonclick()} data-testid="buttonmodal" key={`${todo.id}-edit-button-modal`}><EditOutlined/></Button>,
      ]}
      className="list-item"
      key={todo.id}
      data-testid="listitem"
    >
      <List.Item.Meta
          description={todo.name}
          key={`${todo.id}-meta-key`}
        />
        <Tag key={`${todo.id}-tag-key`} color={todo.priority === 0 ? 'green' : (todo.priority === 1 ? 'orange' : 'red')} className="todo-tag">
          {todo.priority === 0 ? 'Low Priority' : (todo.priority === 1 ? 'Medium Priority' : 'High Priority')}
        </Tag>
      { <Select value={todo.status} disabled={true} key={`${todo.id}-status-select`}>
            {statusList.map((s) => (
              <Option value={s.id} key={`${todo.id}-status-select-option`}>{s.name}</Option>
            ))}
       </Select> }
       <Modal title="Edit Task" visible={modalShow} onCancel={handleCancel} 
              onOk={handleOk} 
              footer={[
                <Button type="primary" onClick={handleCancel}>Close </Button>
              ]}
              key={`${todo.id}-modal`}
              data-testid="editmodal">
          <Form form={form} onFinish={handleOnFinish} 
            initialValues={{
              ["name"]: todo.name,
              ["priority"]: todo.priority,
              ["status"]: todo.status 
             }}
             key={`${todo.id}-form-edit`}
            layout="horizontal" className="todo-form">
             <Row gutter={20}  key={`${todo.id}-rowitem`}>
              <Col xs={24} sm={24} md={15} lg={15} xl={15} key={`${todo.id}-colitem`}>
                    <Form.Item name={'name'} label="Name" key={`${todo.id}-form-item-name`}>
                        <Input  key={`${todo.id}-name-input`}/>
                    </Form.Item>
                    <Form.Item name={'priority'} label="Priority" key={`${todo.id}-form-item-priority`}>
                      <Select key={`${todo.id}-priority-select`}>
                         {priorityList.map((s) => (<Option value={s.id} key={`${todo.id}-priority-select-option`}>{s.name}</Option>))}
                      </Select>
                    </Form.Item>
                    <Form.Item name={'status'} label="Status" key={`${todo.id}-form-item-status`}>
                      <Select key={`${todo.id}-status-selectable`}>
                         {statusList.map((s) => (<Option value={s.id} key={`${todo.id}-status-selectable-option`}>{s.name}</Option>))}
                      </Select>
                    </Form.Item>
              </Col>
              
             </Row>
             <Row key={`${todo.id}-btnrowitem`}> <Button type="primary" htmlType="submit" block key={`${todo.id}-savebtn`}>Save</Button></Row>
          </Form>
        </Modal>
    </List.Item>
  )
}

export default Todo