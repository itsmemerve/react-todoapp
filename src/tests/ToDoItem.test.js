import {render,fireEvent,waitFor } from '@testing-library/react';
import ToDoItem from '../components/ToDoItem';

delete window.matchMedia
window.matchMedia = (query) => ({
  matches: false,
  media: query,
  onchange: null,
  addListener: jest.fn(), // deprecated
  removeListener: jest.fn(), // deprecated
  addEventListener: jest.fn(),
  removeEventListener: jest.fn(),
  dispatchEvent: jest.fn(),
})

// Does Button In a List Item render properly to the screen?
describe("Button In a List Item", () => {
    it("rendered button in list item", () => {
        const{getByTestId} = render(<ToDoItem todo={{id:1, name:"First Task", priority:0, status:0}}/>);
        const frm = getByTestId("buttonmodal");
        expect(frm).toBeTruthy();
    })

    it("rendered list item", () => {
        const{getByTestId} = render(<ToDoItem todo={{id:1, name:"First Task", priority:0, status:0}}/>);
        const frm = getByTestId("listitem");
        expect(frm).toBeTruthy();
    })

    it("click the modal button and see the modal", () => {
        const{baseElement, getByTestId} = render(<ToDoItem todo={{id:1, name:"First Task", priority:0, status:0}}/>);
        const btn = getByTestId("buttonmodal");
        fireEvent.click(btn);
        expect(baseElement).toMatchSnapshot();
        waitFor(() => expect(queryByText('Edit Task')).toBeInTheDocument());
    })
})


