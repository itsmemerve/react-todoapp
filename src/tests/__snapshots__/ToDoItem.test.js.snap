// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button In a List Item click the modal button 1`] = `
<body
  class="ant-scrolling-effect"
  style="overflow: hidden; overflow-x: hidden; overflow-y: hidden;"
>
  <div>
    <li
      class="ant-list-item list-item"
      data-testid="listitem"
    >
      <div
        class="ant-list-item-meta"
      >
        <div
          class="ant-list-item-meta-content"
        >
          <div
            class="ant-list-item-meta-description"
          >
            First Task
          </div>
        </div>
      </div>
      <span
        class="ant-tag ant-tag-green todo-tag"
      >
        Low Priority
      </span>
      <div
        class="ant-select ant-select-single ant-select-show-arrow ant-select-disabled"
      >
        <div
          class="ant-select-selector"
        >
          <span
            class="ant-select-selection-search"
          >
            <input
              aria-activedescendant="rc_select_TEST_OR_SSR_list_0"
              aria-autocomplete="list"
              aria-controls="rc_select_TEST_OR_SSR_list"
              aria-expanded="false"
              aria-haspopup="listbox"
              aria-owns="rc_select_TEST_OR_SSR_list"
              autocomplete="off"
              class="ant-select-selection-search-input"
              disabled=""
              id="rc_select_TEST_OR_SSR"
              readonly=""
              role="combobox"
              style="opacity: 0;"
              type="search"
              unselectable="on"
              value=""
            />
          </span>
          <span
            class="ant-select-selection-item"
            title="Not Started"
          >
            Not Started
          </span>
        </div>
        <span
          aria-hidden="true"
          class="ant-select-arrow"
          style="user-select: none;"
          unselectable="on"
        >
          <span
            aria-label="down"
            class="anticon anticon-down ant-select-suffix"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="down"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
              />
            </svg>
          </span>
        </span>
      </div>
      <ul
        class="ant-list-item-action"
      >
        <li>
          <button
            class="ant-btn ant-btn-primary ant-btn-dangerous remove-todo-button"
            type="button"
          >
            <span>
              X
            </span>
          </button>
          <em
            class="ant-list-item-action-split"
          />
        </li>
        <li>
          <button
            ant-click-animating-without-extra-node="false"
            class="ant-btn ant-btn-default"
            data-testid="buttonmodal"
            type="button"
          >
            <span
              aria-label="edit"
              class="anticon anticon-edit"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="edit"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M257.7 752c2 0 4-.2 6-.5L431.9 722c2-.4 3.9-1.3 5.3-2.8l423.9-423.9a9.96 9.96 0 000-14.1L694.9 114.9c-1.9-1.9-4.4-2.9-7.1-2.9s-5.2 1-7.1 2.9L256.8 538.8c-1.5 1.5-2.4 3.3-2.8 5.3l-29.5 168.2a33.5 33.5 0 009.4 29.8c6.6 6.4 14.9 9.9 23.8 9.9zm67.4-174.4L687.8 215l73.3 73.3-362.7 362.6-88.9 15.7 15.6-89zM880 836H144c-17.7 0-32 14.3-32 32v36c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-36c0-17.7-14.3-32-32-32z"
                />
              </svg>
            </span>
          </button>
        </li>
      </ul>
    </li>
  </div>
  <div>
    <div
      class="ant-modal-root"
      data-testid="editmodal"
    >
      <div
        class="ant-modal-mask"
      />
      <div
        class="ant-modal-wrap"
        tabindex="-1"
      >
        <div
          aria-labelledby="test-id"
          aria-modal="true"
          class="ant-modal"
          role="dialog"
          style="width: 520px;"
        >
          <div
            aria-hidden="true"
            style="width: 0px; height: 0px; overflow: hidden; outline: none;"
            tabindex="0"
          />
          <div
            class="ant-modal-content"
          >
            <button
              aria-label="Close"
              class="ant-modal-close"
              type="button"
            >
              <span
                class="ant-modal-close-x"
              >
                <span
                  aria-label="close"
                  class="anticon anticon-close ant-modal-close-icon"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    data-icon="close"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 00203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
                    />
                  </svg>
                </span>
              </span>
            </button>
            <div
              class="ant-modal-header"
            >
              <div
                class="ant-modal-title"
                id="test-id"
              >
                Edit Task
              </div>
            </div>
            <div
              class="ant-modal-body"
            >
              <form
                class="ant-form ant-form-horizontal todo-form"
              >
                <div
                  class="ant-row"
                  style="margin-left: -10px; margin-right: -10px;"
                >
                  <div
                    class="ant-col ant-col-xs-24 ant-col-sm-24 ant-col-md-15 ant-col-lg-15 ant-col-xl-15"
                    style="padding-left: 10px; padding-right: 10px;"
                  >
                    <div
                      class="ant-form-item"
                    >
                      <div
                        class="ant-row ant-form-item-row"
                      >
                        <div
                          class="ant-col ant-form-item-label"
                        >
                          <label
                            class=""
                            for="name"
                            title="Name"
                          >
                            Name
                          </label>
                        </div>
                        <div
                          class="ant-col ant-form-item-control"
                        >
                          <div
                            class="ant-form-item-control-input"
                          >
                            <div
                              class="ant-form-item-control-input-content"
                            >
                              <input
                                class="ant-input"
                                id="name"
                                type="text"
                                value="First Task"
                              />
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                    <div
                      class="ant-form-item"
                    >
                      <div
                        class="ant-row ant-form-item-row"
                      >
                        <div
                          class="ant-col ant-form-item-label"
                        >
                          <label
                            class=""
                            for="priority"
                            title="Priority"
                          >
                            Priority
                          </label>
                        </div>
                        <div
                          class="ant-col ant-form-item-control"
                        >
                          <div
                            class="ant-form-item-control-input"
                          >
                            <div
                              class="ant-form-item-control-input-content"
                            >
                              <div
                                class="ant-select ant-select-in-form-item ant-select-single ant-select-show-arrow"
                              >
                                <div
                                  class="ant-select-selector"
                                >
                                  <span
                                    class="ant-select-selection-search"
                                  >
                                    <input
                                      aria-activedescendant="priority_list_0"
                                      aria-autocomplete="list"
                                      aria-controls="priority_list"
                                      aria-haspopup="listbox"
                                      aria-owns="priority_list"
                                      autocomplete="off"
                                      class="ant-select-selection-search-input"
                                      id="priority"
                                      readonly=""
                                      role="combobox"
                                      style="opacity: 0;"
                                      type="search"
                                      unselectable="on"
                                      value=""
                                    />
                                  </span>
                                  <span
                                    class="ant-select-selection-item"
                                    title="Low"
                                  >
                                    Low
                                  </span>
                                </div>
                                <span
                                  aria-hidden="true"
                                  class="ant-select-arrow"
                                  style="user-select: none;"
                                  unselectable="on"
                                >
                                  <span
                                    aria-label="down"
                                    class="anticon anticon-down ant-select-suffix"
                                    role="img"
                                  >
                                    <svg
                                      aria-hidden="true"
                                      data-icon="down"
                                      fill="currentColor"
                                      focusable="false"
                                      height="1em"
                                      viewBox="64 64 896 896"
                                      width="1em"
                                    >
                                      <path
                                        d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                                      />
                                    </svg>
                                  </span>
                                </span>
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                    <div
                      class="ant-form-item"
                    >
                      <div
                        class="ant-row ant-form-item-row"
                      >
                        <div
                          class="ant-col ant-form-item-label"
                        >
                          <label
                            class=""
                            for="status"
                            title="Status"
                          >
                            Status
                          </label>
                        </div>
                        <div
                          class="ant-col ant-form-item-control"
                        >
                          <div
                            class="ant-form-item-control-input"
                          >
                            <div
                              class="ant-form-item-control-input-content"
                            >
                              <div
                                class="ant-select ant-select-in-form-item ant-select-single ant-select-show-arrow"
                              >
                                <div
                                  class="ant-select-selector"
                                >
                                  <span
                                    class="ant-select-selection-search"
                                  >
                                    <input
                                      aria-activedescendant="status_list_0"
                                      aria-autocomplete="list"
                                      aria-controls="status_list"
                                      aria-haspopup="listbox"
                                      aria-owns="status_list"
                                      autocomplete="off"
                                      class="ant-select-selection-search-input"
                                      id="status"
                                      readonly=""
                                      role="combobox"
                                      style="opacity: 0;"
                                      type="search"
                                      unselectable="on"
                                      value=""
                                    />
                                  </span>
                                  <span
                                    class="ant-select-selection-item"
                                    title="Not Started"
                                  >
                                    Not Started
                                  </span>
                                </div>
                                <span
                                  aria-hidden="true"
                                  class="ant-select-arrow"
                                  style="user-select: none;"
                                  unselectable="on"
                                >
                                  <span
                                    aria-label="down"
                                    class="anticon anticon-down ant-select-suffix"
                                    role="img"
                                  >
                                    <svg
                                      aria-hidden="true"
                                      data-icon="down"
                                      fill="currentColor"
                                      focusable="false"
                                      height="1em"
                                      viewBox="64 64 896 896"
                                      width="1em"
                                    >
                                      <path
                                        d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                                      />
                                    </svg>
                                  </span>
                                </span>
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="ant-row"
                >
                   
                  <button
                    class="ant-btn ant-btn-primary ant-btn-block"
                    type="submit"
                  >
                    <span>
                      Save
                    </span>
                  </button>
                </div>
              </form>
            </div>
            <div
              class="ant-modal-footer"
            >
              <button
                class="ant-btn ant-btn-primary"
                type="button"
              >
                <span>
                  Close 
                </span>
              </button>
            </div>
          </div>
          <div
            aria-hidden="true"
            style="width: 0px; height: 0px; overflow: hidden; outline: none;"
            tabindex="0"
          />
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Button In a List Item click the modal button and see the modal 1`] = `
<body
  class="ant-scrolling-effect"
  style="overflow: hidden; overflow-x: hidden; overflow-y: hidden;"
>
  <div>
    <li
      class="ant-list-item list-item"
      data-testid="listitem"
    >
      <div
        class="ant-list-item-meta"
      >
        <div
          class="ant-list-item-meta-content"
        >
          <div
            class="ant-list-item-meta-description"
          >
            First Task
          </div>
        </div>
      </div>
      <span
        class="ant-tag ant-tag-green todo-tag"
      >
        Low Priority
      </span>
      <div
        class="ant-select ant-select-single ant-select-show-arrow ant-select-disabled"
      >
        <div
          class="ant-select-selector"
        >
          <span
            class="ant-select-selection-search"
          >
            <input
              aria-activedescendant="rc_select_TEST_OR_SSR_list_0"
              aria-autocomplete="list"
              aria-controls="rc_select_TEST_OR_SSR_list"
              aria-expanded="false"
              aria-haspopup="listbox"
              aria-owns="rc_select_TEST_OR_SSR_list"
              autocomplete="off"
              class="ant-select-selection-search-input"
              disabled=""
              id="rc_select_TEST_OR_SSR"
              readonly=""
              role="combobox"
              style="opacity: 0;"
              type="search"
              unselectable="on"
              value=""
            />
          </span>
          <span
            class="ant-select-selection-item"
            title="Not Started"
          >
            Not Started
          </span>
        </div>
        <span
          aria-hidden="true"
          class="ant-select-arrow"
          style="user-select: none;"
          unselectable="on"
        >
          <span
            aria-label="down"
            class="anticon anticon-down ant-select-suffix"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="down"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
              />
            </svg>
          </span>
        </span>
      </div>
      <ul
        class="ant-list-item-action"
      >
        <li>
          <button
            class="ant-btn ant-btn-primary ant-btn-dangerous remove-todo-button"
            type="button"
          >
            <span>
              X
            </span>
          </button>
          <em
            class="ant-list-item-action-split"
          />
        </li>
        <li>
          <button
            ant-click-animating-without-extra-node="false"
            class="ant-btn ant-btn-default"
            data-testid="buttonmodal"
            type="button"
          >
            <span
              aria-label="edit"
              class="anticon anticon-edit"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="edit"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M257.7 752c2 0 4-.2 6-.5L431.9 722c2-.4 3.9-1.3 5.3-2.8l423.9-423.9a9.96 9.96 0 000-14.1L694.9 114.9c-1.9-1.9-4.4-2.9-7.1-2.9s-5.2 1-7.1 2.9L256.8 538.8c-1.5 1.5-2.4 3.3-2.8 5.3l-29.5 168.2a33.5 33.5 0 009.4 29.8c6.6 6.4 14.9 9.9 23.8 9.9zm67.4-174.4L687.8 215l73.3 73.3-362.7 362.6-88.9 15.7 15.6-89zM880 836H144c-17.7 0-32 14.3-32 32v36c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-36c0-17.7-14.3-32-32-32z"
                />
              </svg>
            </span>
          </button>
        </li>
      </ul>
    </li>
  </div>
  <div>
    <div
      class="ant-modal-root"
      data-testid="editmodal"
    >
      <div
        class="ant-modal-mask"
      />
      <div
        class="ant-modal-wrap"
        tabindex="-1"
      >
        <div
          aria-labelledby="test-id"
          aria-modal="true"
          class="ant-modal"
          role="dialog"
          style="width: 520px;"
        >
          <div
            aria-hidden="true"
            style="width: 0px; height: 0px; overflow: hidden; outline: none;"
            tabindex="0"
          />
          <div
            class="ant-modal-content"
          >
            <button
              aria-label="Close"
              class="ant-modal-close"
              type="button"
            >
              <span
                class="ant-modal-close-x"
              >
                <span
                  aria-label="close"
                  class="anticon anticon-close ant-modal-close-icon"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    data-icon="close"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 00203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
                    />
                  </svg>
                </span>
              </span>
            </button>
            <div
              class="ant-modal-header"
            >
              <div
                class="ant-modal-title"
                id="test-id"
              >
                Edit Task
              </div>
            </div>
            <div
              class="ant-modal-body"
            >
              <form
                class="ant-form ant-form-horizontal todo-form"
              >
                <div
                  class="ant-row"
                  style="margin-left: -10px; margin-right: -10px;"
                >
                  <div
                    class="ant-col ant-col-xs-24 ant-col-sm-24 ant-col-md-15 ant-col-lg-15 ant-col-xl-15"
                    style="padding-left: 10px; padding-right: 10px;"
                  >
                    <div
                      class="ant-form-item"
                    >
                      <div
                        class="ant-row ant-form-item-row"
                      >
                        <div
                          class="ant-col ant-form-item-label"
                        >
                          <label
                            class=""
                            for="name"
                            title="Name"
                          >
                            Name
                          </label>
                        </div>
                        <div
                          class="ant-col ant-form-item-control"
                        >
                          <div
                            class="ant-form-item-control-input"
                          >
                            <div
                              class="ant-form-item-control-input-content"
                            >
                              <input
                                class="ant-input"
                                id="name"
                                type="text"
                                value="First Task"
                              />
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                    <div
                      class="ant-form-item"
                    >
                      <div
                        class="ant-row ant-form-item-row"
                      >
                        <div
                          class="ant-col ant-form-item-label"
                        >
                          <label
                            class=""
                            for="priority"
                            title="Priority"
                          >
                            Priority
                          </label>
                        </div>
                        <div
                          class="ant-col ant-form-item-control"
                        >
                          <div
                            class="ant-form-item-control-input"
                          >
                            <div
                              class="ant-form-item-control-input-content"
                            >
                              <div
                                class="ant-select ant-select-in-form-item ant-select-single ant-select-show-arrow"
                              >
                                <div
                                  class="ant-select-selector"
                                >
                                  <span
                                    class="ant-select-selection-search"
                                  >
                                    <input
                                      aria-activedescendant="priority_list_0"
                                      aria-autocomplete="list"
                                      aria-controls="priority_list"
                                      aria-haspopup="listbox"
                                      aria-owns="priority_list"
                                      autocomplete="off"
                                      class="ant-select-selection-search-input"
                                      id="priority"
                                      readonly=""
                                      role="combobox"
                                      style="opacity: 0;"
                                      type="search"
                                      unselectable="on"
                                      value=""
                                    />
                                  </span>
                                  <span
                                    class="ant-select-selection-item"
                                    title="Low"
                                  >
                                    Low
                                  </span>
                                </div>
                                <span
                                  aria-hidden="true"
                                  class="ant-select-arrow"
                                  style="user-select: none;"
                                  unselectable="on"
                                >
                                  <span
                                    aria-label="down"
                                    class="anticon anticon-down ant-select-suffix"
                                    role="img"
                                  >
                                    <svg
                                      aria-hidden="true"
                                      data-icon="down"
                                      fill="currentColor"
                                      focusable="false"
                                      height="1em"
                                      viewBox="64 64 896 896"
                                      width="1em"
                                    >
                                      <path
                                        d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                                      />
                                    </svg>
                                  </span>
                                </span>
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                    <div
                      class="ant-form-item"
                    >
                      <div
                        class="ant-row ant-form-item-row"
                      >
                        <div
                          class="ant-col ant-form-item-label"
                        >
                          <label
                            class=""
                            for="status"
                            title="Status"
                          >
                            Status
                          </label>
                        </div>
                        <div
                          class="ant-col ant-form-item-control"
                        >
                          <div
                            class="ant-form-item-control-input"
                          >
                            <div
                              class="ant-form-item-control-input-content"
                            >
                              <div
                                class="ant-select ant-select-in-form-item ant-select-single ant-select-show-arrow"
                              >
                                <div
                                  class="ant-select-selector"
                                >
                                  <span
                                    class="ant-select-selection-search"
                                  >
                                    <input
                                      aria-activedescendant="status_list_0"
                                      aria-autocomplete="list"
                                      aria-controls="status_list"
                                      aria-haspopup="listbox"
                                      aria-owns="status_list"
                                      autocomplete="off"
                                      class="ant-select-selection-search-input"
                                      id="status"
                                      readonly=""
                                      role="combobox"
                                      style="opacity: 0;"
                                      type="search"
                                      unselectable="on"
                                      value=""
                                    />
                                  </span>
                                  <span
                                    class="ant-select-selection-item"
                                    title="Not Started"
                                  >
                                    Not Started
                                  </span>
                                </div>
                                <span
                                  aria-hidden="true"
                                  class="ant-select-arrow"
                                  style="user-select: none;"
                                  unselectable="on"
                                >
                                  <span
                                    aria-label="down"
                                    class="anticon anticon-down ant-select-suffix"
                                    role="img"
                                  >
                                    <svg
                                      aria-hidden="true"
                                      data-icon="down"
                                      fill="currentColor"
                                      focusable="false"
                                      height="1em"
                                      viewBox="64 64 896 896"
                                      width="1em"
                                    >
                                      <path
                                        d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                                      />
                                    </svg>
                                  </span>
                                </span>
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="ant-row"
                >
                   
                  <button
                    class="ant-btn ant-btn-primary ant-btn-block"
                    type="submit"
                  >
                    <span>
                      Save
                    </span>
                  </button>
                </div>
              </form>
            </div>
            <div
              class="ant-modal-footer"
            >
              <button
                class="ant-btn ant-btn-primary"
                type="button"
              >
                <span>
                  Close 
                </span>
              </button>
            </div>
          </div>
          <div
            aria-hidden="true"
            style="width: 0px; height: 0px; overflow: hidden; outline: none;"
            tabindex="0"
          />
        </div>
      </div>
    </div>
  </div>
</body>
`;
